File: Silver II/BOJ_1929/BOJ_1929.cpp
#include <iostream>
#include <vector>

using namespace std;

int n, m;
vector<bool> isPrime(1000001, true);

int main()
{
    cin >> n >> m;

    isPrime[1] = false;
    for (int i = 2; i <= m; ++i) {
        if (!isPrime[i]) {
            continue;
        }
        for (int j = i * 2; j <= m; j += i) {
            isPrime[j] = false;
        }
    }

    for (int i = n; i <= m; ++i) {
        if (isPrime[i]) {
            cout << i << '\n';
        }
    }

    return 0;
}

File: Silver II/BOJ_1929/BOJ_1929.py
import sys


def init():
    isPrime = [True for _ in range(1000001)]
    isPrime[0] = isPrime[1] = False

    for i in range(1000001):
        if isPrime[i]:
            for j in range(i * i, 1000001, i):
                isPrime[j] = False

    return isPrime


is_prime = init()
n, m = list(map(int, sys.stdin.readline().split()))
# print(n, m)

for i in range(n, m + 1):
    if is_prime[i]:
        print(i)
